transport/src/main/java/io/netty/channel/ChannelOutboundInvoker.java

```diff
@@ -119,11 +167,15 @@
      * The given {@link ChannelPromise} will be notified.
      * <p>
      * This will result in having the
-     * {@link ChannelHandler#bind(ChannelHandlerContext, SocketAddress, ChannelPromise)} method
+     * {@link ChannelHandler#bind(ChannelHandlerContext, SocketAddress, ChannelOutboundInvokerCallback)} method
      * called of the next {@link ChannelHandler} contained in the {@link ChannelPipeline} of the
      * {@link Channel}.
+     *
+     * @param localAddress      the {@link SocketAddress} to which it should bound
+     * @param callback  the {@link ChannelOutboundInvokerCallback} to notify once the operation completes
+     * @return itself.
      */
-    ChannelFuture bind(SocketAddress localAddress, ChannelPromise promise);
+    I bind(SocketAddress localAddress, ChannelOutboundInvokerCallback callback);
```

Comment:
We could also consider adding the methods that take `ChannelOutboundInvokerCallback` only to `ChannelHandlerContext` as this is usually the most useful place to use these. This would "minimise" the API surface on `Channel`.